### PR TITLE
Course Show Page

### LINKED ISSUE #10: Create Course Show Page
# User Story

* As an admin or instructor
* I can navigate from the /courses page to a specific page for each course
* So that I can see a page that has just information about my specific course, e.g. place to upload student roster, place to add student or staff etc.  (Initially this can just show a course table like the one on the /courses page, but with just one row for the specific course.)

# Discussion

We need a page for each course where we can put things such as 
* Course roster upload link
* Staff roster
* etc

In particular, we need a place for the student roster upload, and a place for StudentTable to show up.  (Those are separate issues).

Your job is simple: build a page that looks a lot like the course page, but it will have the url `/courses/{:id}`.   It will only retrieve one row from the database for the course with id=:id, and show just that one course.

That's it for this issue; then once this is built, we can add many other things.

# Acceptance Criteria

- [ ] From the /courses page there is a way to navigate to each /courses/:id page, either by a hyperlink (e.g. making the course id or name a hyperlink) or via a button; this is implementers choice.
- [ ] When you click that button, you are taken to a page such as /courses/1, /courses/2, etc. that looks just like /courses page (for now) but only has one course shown in the table.
- [ ] The courses page reuses the CourseTable React component.
- [ ] The courses show page is a different page component from CoursesIndex, called CoursesShow, because we will eventually be putting lots of different stuff on it.

### PR DESCRIPTION
**Pull request by Peak1260**
In this pull request, I implemented a course show page that will allow admins and instructors to have a look at individual course information along with its corresponding tests and storybooks. I created a show button which will link you to that page. There is also a short description to let the admins/instructors know what to do on this page, and links created to add students, staff, or class rosters though those are not linked to anything yet as it is for future implementation. 

Dokku deployment:
https://proj1-peak1260-dev.dokku-09.cs.ucsb.edu/

Closes #10.

### CODE REVIEW

File: frontend/src/main/pages/CoursesShowPage.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import CoursesTable from 'main/components/Courses/CoursesTable';
+import { useBackend } from 'main/utils/useBackend';
+import { useCurrentUser } from 'main/utils/currentUser';
+
+export default function CoursesShowPage() {
+    let { id } = useParams();
+    const { data: currentUser } = useCurrentUser();
+
+    const { data: courses, error: _error, status: _status } =
+        useBackend(
```
Comment (pconrad) [inline frontend/src/main/pages/CoursesShowPage.js:15]:
```suggestion
       // Stryker disable all 
        useBackend(
```

File: frontend/src/main/pages/CoursesShowPage.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            [`/api/courses?id=${id}`],
+            {
+                method: "GET", url: "/api/courses/get",
+                params: {
+                    id
+                },
+            },
+    []
+        );
```
Comment (pconrad) [inline frontend/src/main/pages/CoursesShowPage.js:24]:
```suggestion
        );
        // Stryker restore all
```

File: frontend/src/main/pages/CoursesShowPage.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            checkLength = [courses];
+        } else if (courses) {
+            checkLength = courses;
+        }
+
+    return (
+        <BasicLayout>
+            <div className="pt-2">
+                <h1>Individual Course Information</h1>
+                <CoursesTable courses={checkLength} currentUser={currentUser} />
```
Comment (pconrad) [inline frontend/src/main/pages/CoursesShowPage.js]:
```suggestion
                <CoursesTable courses={course ? [course] : []} currentUser={currentUser} />
```

This says: if you have one course, put it in an array which is what the CoursesTable expects.  If you don't have one course, then just display an empty list.

**Review by pconrad — COMMENTED** [review]
A few suggestions

**Comment by rtzhang1x** [comment]
I should probably leave that commit in a seperate branch...
https://github.com/ucsb-cs156-s24/proj-organic-s24-5pm-1/pull/24

**Comment by sanilkatula** [comment]
It's good.

One suggestion - when we go to the Show page lets keep the info limited. 

Screenshot 2024-05-28 at 6 04 18 PM

instead of show, staff, edit, delete, etc all these button. 

Let's stay with the edit button only. And at max the staff button because the below upload staff is not implemented.

**Review by sanilkatula — CHANGES_REQUESTED** [review]
see comment

**Review by sanilkatula — DISMISSED** [review]

**Comment by sanilkatula** [comment]
LGTM

**Review by GeorgebigG — DISMISSED** [review]
LGTM!

**Review by pconrad — CHANGES_REQUESTED** [review]
Please remove all links to pages that are not yet implemented, or else put in placeholder pages.  

We don't merge code into main with buttons or links that lead to error messages. That's not good practice.  The main branch shouldn't have known bugs in it.

**Comment by Peak1260** [comment]
Hyperlinks are now removed.

**Review by JUstISAGE — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM.   Staff button still leads to a non-existing page, but I know it's coming so I'll let it slide.

**Pull request merged**